src/utils/filterTypes.ts

```diff
@@ -0,0 +1,51 @@
+export type Filter = {
+  campo: string;
+  operador: string;
+  valor: string | number | string[];
+};
+
+export type Sorter = {
+  campo: string;
+  ordem: 'ASC' | 'DESC';
+};
+
+export function buildWhereClause(filters: Filter[]) {
+  const where: Record<string, unknown> = {};
+
+  if (!filters || filters.length === 0) {
+    return where;
+  }
```

Comment:
podemos definir o valor padrão de filters como um array vazio `[]`, eliminando a necessidade do if?
```suggestion
export function buildWhereClause(filters: Filter[] = []) {
  const where: Record<string, unknown> = {};
```